server/common/utils/helpers.js

```diff
@@ -388,7 +388,8 @@ const runAsyncTasks = async (...tasks) => {
 /* eslint-disable-next-line no-console */
 const fireAndForget = promise => promise.catch(err => console.error(err));
 
-const returnPayload = promise => payload => promise.then(() => payload);
+const returnPayload = (promise, returnResult = false) => payload =>
+  promise.then(result => (returnResult ? { payload, result } : payload));
```

Comment:
```
const returnPayload = promise => (promiseResult = false) =>  payload =>
  promise.then(result => (returnResult ? { payload, result } : payload));
```

Reply:
I restored to the previous version because we will be replacing this method with simple `thens` or `async`-`awaits`